### PR TITLE
Add Extension documentation

### PR DESCRIPTION
**Pull request by Aleh-Yanushkevich**

### CODE REVIEW

File: docs/Extension.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+public class ReportPortalEventsObserver : IReportEventsObserver
+{
+    public void Initialize(IReportEventsSource reportEventsSource)
+    {
+        reportEventsSource.OnBeforeTestFinishing += ReportEventsSource_OnBeforeTestFinishing;
+    }
+
+    private void ReportEventsSource_OnBeforeTestFinishing(ITestReporter testReporter, BeforeTestFinishingEventArgs args)
+    {
+        var methodInfo = TestExecutionContext.CurrentContext.CurrentTest.Method;
```
Comment (nvborisenko) [inline docs/Extension.md:30]: Example should be independent on unit test framework. BTW this example will not even work.

**Review by nvborisenko — CHANGES_REQUESTED** [review]

File: docs/Extension.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
+# Extension
```
Comment (nvborisenko) [inline docs/Extension.md:1]: I think `Extensibility.md` file name is more generic.

File: docs/Extension.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+```
+
+### What can be observed?
+`IReportEventsObserver` interface allows to observe next events:
+
+- Before launch starting
+- Launch initializing
+- After launch starting
+- Before test starting
+- After test started
```
Comment (nvborisenko) [inline docs/Extension.md:90]: Let's get rid off most of these events to have weak reference. It allows us to make code changes without updating documentation.

File: docs/Extension.md (hunk 1 above)
Comment (Aleh-Yanushkevich) [reply docs/Extension.md]: Updated by replacing on more generic example.

File: docs/Extension.md (hunk 2 above)
Comment (Aleh-Yanushkevich) [reply docs/Extension.md]: Renamed.

File: docs/Extension.md (hunk 3 above)
Comment (Aleh-Yanushkevich) [reply docs/Extension.md]: Fixed.

**Review by nvborisenko — APPROVED** [review]

**Pull request merged**